automatic troubleshooting endpoint context docs (#728)
elastic/endpoint-package: package/endpoint/docs/knowledge_base/endpoint_exceptions/false_positive_malware_detection.md

- **Memory signature** (in-memory YARA scanning): Creates alerts with `event.code: memory_signature`. The engine scans executable memory regions for known threat signatures. Fix with an **Endpoint Alert Exception** on the specific memory signature rule and process.
- **Blocklist**: A separate artifact that prevents known-malicious binaries from executing. It is NOT the inverse of an exception — blocklist entries block execution, while exceptions suppress alerts. If a process matches the Global Exception List (e.g. signed by a trusted vendor like Microsoft), the User Blocklist is not evaluated, which can cause blocklist entries to appear ineffective for certain trusted-signed binaries.

Key distinction: An **Endpoint Alert Exception** suppresses alerts while keeping monitoring active. A **Trusted Application** stops Endpoint from monitoring the process entirely. For false positives, an Endpoint Alert Exception is almost always the correct choice.

## Common issues

### Malware engine false positive on legitimate software

> The malware ML model assigns a score to a file's binary content. Legitimate software can occasionally score above the threshold (default 0.58), especially newly released or niche software that the model has not been trained on. The alert's `file.Ext.malware_classification.score` and `file.Ext.malware_classification.threshold` fields show the model's confidence.

Reviewer: The exact value changes every time the model is changed

```suggestion
The malware ML model assigns a score to a file's binary content. Legitimate software can occasionally score above the threshold, especially newly released or niche software that the model has not been trained on. The alert's `file.Ext.malware_classification.score` and `file.Ext.malware_classification.threshold` fields show the model's confidence.
```

3. **Check for other security software**: Microsoft Defender or other AV running alongside Elastic Defend may independently block the file. Correlate the timing — if the file is blocked before an endpoint alert appears, another product may be responsible.

Review the full policy configuration in the agent diagnostics or via `get_package_configurations` to identify which protections are enabled and in which mode.

### Behavioral rule false positive

Behavioral detection rules monitor runtime patterns across the system. A false positive occurs when legitimate software performs actions that match a malicious pattern (e.g. ransomware-like file encryption, unusual DLL loading via `rundll32.exe`).

The alert's `kibana.alert.rule.name` or `rule.description` field identifies which behavioral rule fired. Common false-positive behavioral rules include:
- "Unusual DLL Extension Loaded by Rundll32 or Regsvr32" — can fire during legitimate software installations that use `rundll32.exe` to load DLLs from temporary directories.
- "Suspicious PowerShell Execution" — can fire for legitimate IT management scripts.

> To suppress on Elastic Defend 9.2+: add the process as a **Trusted Application** (this disables behavioral detections for the process entirely) OR add an **Endpoint Alert Exception** targeting the specific rule ID and process if you want to suppress only that one rule.

Reviewer: ```suggestion
To suppress: add an **Endpoint Alert Exception** targeting the specific rule ID and process if you want to suppress only that one rule.
```

> To suppress on pre-9.2: Trusted Applications do not suppress behavioral detections on these older versions. Use an **Endpoint Alert Exception** targeting `rule.id` and the relevant process fields.

Reviewer: ```suggestion
```

### Memory signature false positive

Memory signature alerts (`event.code: memory_signature`) occur when YARA rules match patterns found in executable memory regions. Rare false positives can occur when:
- The endpoint's own process memory contains YARA signature strings (because the signatures themselves are loaded into memory for matching).
- A third-party driver or application causes memory corruption that makes non-executable memory pages appear executable, leading to unexpected scanning of heap regions.

> - Faulty RAM causes bit flips in page table entries, changing page protection bits and exposing non-code memory to the scanner.

Reviewer: ```suggestion
```

For a one-off suppression: create an Endpoint Alert Exception using `event.code: memory_signature` AND `process.executable` matching the affected process.

### Alert field data corrupted by product bug

In rare cases, the endpoint may populate alert fields with corrupted data (e.g. truncated `dll.path` or `dll.name` values due to kernel-level memory corruption from a third-party driver). When the corrupted field values happen to match a behavioral detection rule's criteria — for example, a DLL path truncated to appear as if it has an unusual extension — the rule fires on data that does not reflect the actual system state.

These are product bugs, not fixable via exceptions in the general case. However, a targeted Endpoint Alert Exception can suppress the specific false-positive pattern as a workaround. For example, suppressing a rule when the DLL's code signature status indicates it was never actually loaded (`dll.code_signature.status: "errorCode_endpoint: Initital state, no attempt to load signature was made"`).

> If you encounter alerts where field values appear truncated or nonsensical, collect agent diagnostics and the full alert document for Elastic Support. Check for third-party kernel drivers that may be causing memory corruption — the Windows Driver Verifier can help identify misbehaving drivers on non-production systems.

Reviewer: This was a very edge case, I think we should remove it. It won't generalize well.
```suggestion
```

### Blocklist entry not blocking a trusted-signed process

The Elastic Defend blocklist prevents known-malicious binaries from executing. However, if a process matches the **Global Exception List** before the blocklist is evaluated, the blocklist entry is skipped. The Global Exception List contains entries for major trusted software publishers (e.g. Microsoft). This means binaries signed by these vendors cannot be blocked via the user blocklist — the global exception takes priority.

> This is a known product limitation. If the goal is to prevent specific trusted-signed binaries from running, the blocklist is not the correct mechanism. Consider using operating system-level application control policies (e.g. Windows AppLocker or WDAC) instead.

Reviewer: ```suggestion
```

## Investigation priorities

> 1) Determine the detection engine that generated the alert by checking `event.code`: `malicious_file` (malware engine), `behavior` (behavioral protection), `memory_signature` (memory scanning). This determines which artifact type to use for suppression.

Reviewer: ```suggestion
1) Determine the detection engine that generated the alert by checking `event.code`: `malicious_file` (malware protection), `behavior` (malicious behavior protection), `memory_signature` (memory threat protection), `ransomware` (ransomware protection). This determines which artifact type to use for suppression.
```
